### PR TITLE
Dana/hf-alb-docs

### PR DESCRIPTION
**Pull request by danaaubakirova**

### CODE REVIEW

File: doc_aug_hf_alb.md [hunk 1]
```diff
@@ -0,0 +1,476 @@
...
+    Requirement already satisfied: regex>=2021.8.3 in /fsx/dana_aubakirova/miniconda/envs/myenv/lib/python3.11/site-packages (from nltk) (2023.12.25)
+    Requirement already satisfied: tqdm in /fsx/dana_aubakirova/miniconda/envs/myenv/lib/python3.11/site-packages (from nltk) (4.65.0)
+
+
+
+```python
+import nltk
+
+nltk.download('stopwords')
+from nltk.corpus import stopwords
```
Comment (andimarafioti) [inline doc_aug_hf_alb.md:242]: More preparation which I would handle at the start

File: doc_aug_hf_alb.md [hunk 2]
```diff
@@ -0,0 +1,476 @@
...
+
+```python
+transformed = transform(image=image, textimage_metadata=metadata)
+```
+
+
+```python
+visualize(transformed['image'])
+```
+
```
Comment (andimarafioti) [inline doc_aug_hf_alb.md:269]: all of this seems like something to explain. I know nltk but for someone who doesn't this code is hard to follow.

File: doc_aug_hf_alb.md [hunk 3]
```diff
@@ -0,0 +1,476 @@
...
+
+```python
+transformed = transform_complex(image=image, textimage_metadata=metadata)
+```
+
+
+```python
+visualize(transformed["image"])
+```
```
Comment (andimarafioti) [inline doc_aug_hf_alb.md:294]: more things to explain, and code to put together. Just having a title is not enough 😅

File: doc_aug_hf_alb.md [hunk 4]
```diff
@@ -0,0 +1,476 @@
...
+     {'bbox_coords': (562, 583, 1987, 631),
+      'text': 'PUBLIC MEETING ON PRE-MARKET is hasn EVALUATION OF than',
+      'original_text': 'PUBLIC MEETING ON PRE-MARKET EVALUATION OF',
+      'bbox_index': 7,
+      'font_color': 'red'},
+     {'bbox_coords': (525, 2375, 2173, 2421),
+      'text': "Emphasized that it couldn don't won't is critical that the she generic shouldn product is weren no less abuse-deterrent",
+      'original_text': 'Emphasized that it is critical that the generic product is no less abuse-deterrent',
+      'bbox_index': 26,
+      'font_color': 'red'}]
```
Comment (andimarafioti) [inline doc_aug_hf_alb.md:409]: This is just printed as text. Are you sure that you want to do that? How will it look like in the blog?

File: doc_aug_hf_alb.md [hunk 5]
```diff
@@ -0,0 +1,476 @@
...
+
+
+```python
+transformed = transform(image=image_template, textimage_metadata=metadata)
+```
+
+
+```python
+visualize(transformed['image'])
+```
```
Comment (andimarafioti) [inline doc_aug_hf_alb.md:447]: all together

File: doc_aug_hf_alb.md [hunk 6]
```diff
@@ -0,0 +1,476 @@
...
+
+    
+![png](assets/185_albumentations/output_46_0.png)
+    
+
+
+
+```python
+
+```
```
Comment (andimarafioti) [inline doc_aug_hf_alb.md:458]: empty python blocK?

Comment (danaaubakirova) [reply doc_aug_hf_alb.md:458]: removed

File: doc_aug_hf_alb.md [hunk 7]
```diff
@@ -0,0 +1,476 @@
...
+
+## Introduction
+
+To address the need, we are introducing a new data augmentation pipeline developed in collaboration with Albumentations AI. This pipeline is designed to handle both images and text within them, providing a comprehensive solution for document images. We refer to this type of data augmentation as *multimodal*, as it modifies both the image content and the text annotations.
+
+As discussed in a previous [blog post](https://huggingface.co/blog/danaaubakirova/doc-augmentation), our goal is to test the hypothesis that integrating augmentations on both text and images during pretraining is effective. Detailed parameters and use case illustrations can be found on the  [Albumentations AI Documentation](https://albumentations.ai/docs/examples/example_textimage/?h=textimage). Albumentations AI enables the dynamic design of these augmentations and their integration with other types of augmentations.
+
+# Method 
+
+To augment document images, we begin by randomly selecting lines within the document. A hyperparameter `fraction_range` controls the fraction of bounding box to be modified.
```
Comment (molbap) [inline doc_aug_hf_alb.md:29]:
```suggestion
To augment document images, we begin by randomly selecting lines within the document. A hyperparameter `fraction_range` controls the bounding box fraction to be modified.
```

File: doc_aug_hf_alb.md (hunk 2 above)
Comment (danaaubakirova) [reply doc_aug_hf_alb.md]: Good point. I tried to explain what are stop words.

File: doc_aug_hf_alb.md [hunk 8]
```diff
@@ -0,0 +1,476 @@
...
+
+The need for effective data augmentation techniques for document images became evident when addressing challenges in fine-tuning models with limited datasets. A common concern is that typical image transformations, such as resizing, blurring, or changing background colors, can negatively impact text extraction accuracy.
+
+![png](assets/185_albumentations/quote2.png)
+
+We recognized the need for data augmentation techniques that preserve the integrity of the text while augmenting the dataset. Such data augmentation can facilitate the easy generation of new documents or the modification of existing ones, while preserving the quality of the text.
+
+## Introduction
+
+To address the need, we are introducing a new data augmentation pipeline developed in collaboration with Albumentations AI. This pipeline is designed to handle both images and text within them, providing a comprehensive solution for document images. We refer to this type of data augmentation as *multimodal*, as it modifies both the image content and the text annotations.
```
Comment (molbap) [inline doc_aug_hf_alb.md:23]: Also maybe insert link to albumentations here? 
```suggestion
To address this need, we introduce a **new data augmentation pipeline** developed in collaboration with Albumentations AI. This pipeline handles both images and text within them, providing a comprehensive solution for document images. This class of data augmentation is *multimodal* as it modifies both the image content and the text annotations simultaneously.
```

File: doc_aug_hf_alb.md [hunk 9]
```diff
@@ -0,0 +1,476 @@
...
+
+In this blog post, we provide a tutorial on how to use a new data augmentation technique for document images, developed in collaboration with Albumentations AI.
+
+## Motivation
+
+The need for effective data augmentation techniques for document images became evident when addressing challenges in fine-tuning models with limited datasets. A common concern is that typical image transformations, such as resizing, blurring, or changing background colors, can negatively impact text extraction accuracy.
+
+![png](assets/185_albumentations/quote2.png)
+
+We recognized the need for data augmentation techniques that preserve the integrity of the text while augmenting the dataset. Such data augmentation can facilitate the easy generation of new documents or the modification of existing ones, while preserving the quality of the text.
```
Comment (molbap) [inline doc_aug_hf_alb.md:19]:
```suggestion
We recognized the need for data augmentation techniques that preserve the integrity of the text while augmenting the dataset. Such data augmentation can facilitate generation of new documents or modification of existing ones, while preserving their text quality.
```

File: doc_aug_hf_alb.md [hunk 10]
```diff
@@ -0,0 +1,476 @@
...
+
+## Introduction
+
+To address the need, we are introducing a new data augmentation pipeline developed in collaboration with Albumentations AI. This pipeline is designed to handle both images and text within them, providing a comprehensive solution for document images. We refer to this type of data augmentation as *multimodal*, as it modifies both the image content and the text annotations.
+
+As discussed in a previous [blog post](https://huggingface.co/blog/danaaubakirova/doc-augmentation), our goal is to test the hypothesis that integrating augmentations on both text and images during pretraining is effective. Detailed parameters and use case illustrations can be found on the  [Albumentations AI Documentation](https://albumentations.ai/docs/examples/example_textimage/?h=textimage). Albumentations AI enables the dynamic design of these augmentations and their integration with other types of augmentations.
```
Comment (molbap) [inline doc_aug_hf_alb.md:25]:
```suggestion
As discussed in a previous [blog post](https://huggingface.co/blog/danaaubakirova/doc-augmentation), our goal is to test the hypothesis that integrating augmentations on both text and images during pretraining of VLMs is effective. Detailed parameters and use case illustrations can be found on the  [Albumentations AI Documentation](https://albumentations.ai/docs/examples/example_textimage/?h=textimage). Albumentations AI enables the dynamic design of these augmentations and their integration with other types of augmentations.
```

File: doc_aug_hf_alb.md (hunk 3 above)
Comment (danaaubakirova) [reply doc_aug_hf_alb.md]: Added explanation.

File: doc_aug_hf_alb.md [hunk 11]
```diff
@@ -0,0 +1,476 @@
...
+- user: Ternaus
+guest: True
+---
+# Introducing Multimodal TextImage Augmentation for Document Images
+
+In this blog post, we provide a tutorial on how to use a new data augmentation technique for document images, developed in collaboration with Albumentations AI.
+
+## Motivation
+
+The need for effective data augmentation techniques for document images became evident when addressing challenges in fine-tuning models with limited datasets. A common concern is that typical image transformations, such as resizing, blurring, or changing background colors, can negatively impact text extraction accuracy.
```
Comment (molbap) [inline doc_aug_hf_alb.md:15]: I'd try to recontextualize a bit the posting wrt to the wave of VLMs:
```suggestion
Visual Language Models (VLMs) have an immense range of application, but they often need to be fine-tuned to a range of specific use-cases, in particular for datasets containing document images, i.e., images with high textual content. And we want a model to learn to read properly, which is made difficult in the most frequent of cases: where data is missing.

Hence, the need for **effective data augmentation** techniques for document images became evident when addressing challenges in fine-tuning models with limited datasets. A common concern is that typical image transformations, such as resizing, blurring, or changing background colors, can negatively impact text extraction accuracy.
```

**Review by molbap — COMMENTED** [review]
A few comments here and there, mostly trying to recontextualize the beginning

File: doc_aug_hf_alb.md (hunk 4 above)
Comment (molbap) [reply doc_aug_hf_alb.md]: Maybe just one element of this dictionary is enough - no need for it to be copy-pastable either, so an image containing that would more readable/enticing to read, for instance on the left the modified image cropped to the line of interest, and on the right an image of the dictionary entry of interest?

File: doc_aug_hf_alb.md (hunk 11 above)
Comment (danaaubakirova) [reply doc_aug_hf_alb.md:15]: Yes, I think this is a good point. It is better when the text-richness of documents and their tole in fine-tuning and pertaining VLMs is clearly articulated.

File: doc_aug_hf_alb.md (hunk 8 above)
Comment (danaaubakirova) [reply doc_aug_hf_alb.md]: added the link

**Review by andimarafioti — CHANGES_REQUESTED** [review]
Sorry, I didn't see in the first pass, but you're not supposed to upload images to this repo (except the thumbnail). Please add the images to a dataset like we did for the last blog and load them with that. Also, rewrite the git history so that the images don't get uploaded to main (otherwise loading the repo takes forever)

**Review by andimarafioti — APPROVED** [review]
Thank you for addressing the comments!

**Review by andimarafioti — APPROVED** [review]
let's gooooo
